### ISSUE TITLE
SAML: OpenSamlMetadataResolver doesn't include NameID formats in output

### ISSUE THREAD

**Issue by handcraftedbits** [spring-projects/spring-security#9297]
**Expected Behavior**

`OpenSamlMetadataResolver` should serialize NameID formats associated with an SP.

**Current Behavior**

`OpenSamlMetadataResolver` does not serialize any NameID formats.

**Context**

It is typical for SP descriptors to include a set of supported NameID formats.  `OpenSamlMetadataResolver` does not even attempt to serialize a default set of them, let alone user-provided ones.  This may be partly due to `RelyingPartyRegistration` not allowing users to define them.  At a minimum, `OpenSamlMetadataResolver` should attempt to include the formats in the `urn:oasis:names:tc:SAML:1.1:nameid-format` namespace.  It might also be helpful for `OpenSamlMetadataResolver` not to be marked `final` so users can customize the serizliation.

**Comment by jzheaux** [comment]
Related to #9115

**Comment by ctmay4** [comment]
I am having the same issue.  The error is misleading:

```
[subject_not_found] Assertion [_a79ccbf3-6309-4f18-a132-2a4c91653a19] is missing a subject
```

However when you dig into the code and review the response the reason is the NameID is not returned.  Did you ever find a workaround?

**Comment by jzheaux** [comment]
Closed in a17dfb8456cf3a517d44cfcccdaa7a1d2ae6121f
